pandas/core/arrays/base.py

```diff
@@ -705,7 +708,11 @@ def argsort(
         #    it is recommended to also override argmax/argmin
         ascending = nv.validate_argsort_with_ascending(ascending, args, kwargs)
 
-        values = self._values_for_argsort()
+        # TODO: ENH argsort for SparseArray
+        if type(self).__name__ == "SparseArray":
+            values = ExtensionArray._values_for_argsort(self)
+        else:
+            values = self._values_for_argsort()
```

Comment:
Instead of adding this workaround, what about just directly handling this in `SparseArray` by accessing `_sparse_values`?